src/main/java/com/nanalysis/spinlab/dataset/values/ListNumberValue.java

```diff
@@ -85,6 +87,15 @@ public String stringValue() {
         return value.stream().map(String::valueOf).collect(Collectors.joining(";"));
     }
 
+    public List<Double> getValueAsHertz(Header header) {
+        Unit unit = numberType != null ? numberType.getUnit() : null;
+        if (unit == Unit.Hertz) {
+            return doubleListValue();
+        }
+
+        throw new IllegalStateException("Trying read a Hertz value from unit type " + unit);
+    }
```

Comment:
No unit notion, so every ListNumber value for every type will have a method called getValueAsHertz? That seem very specific to the HzPpm subclass.
Wouldn't it be more general for such a class to have a ``getValueAs(Unit unit, Header header)``. For it can onlt handle the current unit, but there are easy conversiont hat could be performed ingeneral, e.g. from milli to micro of this kind of stuff.

Reply:
Good idea, changed it as you suggested.